sycl/source/detail/os_util.cpp

```diff
@@ -66,9 +71,25 @@ OSModuleHandle OSUtil::getOSModuleHandle(const void *VirtAddr) {
 }
 
 #elif defined(SYCL_RT_OS_WINDOWS)
+// TODO: implement this function for Windows probably by using
 // GetModuleHandleEx(GET_MODULE_HANDLE_EX_FLAG_UNCHANGED_REFCOUNT,...)
-// to implement getOSModuleHandle
-#endif // SYCL_RT_OS_LINUX
+OSModuleHandle OSUtil::getOSModuleHandle(const void *VirtAddr) {
+  throw runtime_error("OSUtil::getOSModuleHandle() is not implemented yet");
+}
+#endif // SYCL_RT_OS_WINDOWS
+
+size_t OSUtil::getOSMemSize() {
+#if defined(SYCL_RT_OS_LINUX)
+  struct sysinfo MemInfo;
```

Comment:
In C++ probably
```
   sysinfo MemInfo;
```
is enough.

Reply:
In this case 'struct' keyword is needed. Otherwise the error is emitted.

Reply:
Ah, very curious. 
Perhaps because in `int sysinfo(struct sysinfo *info);` `sysinfo` is the name of a function and the name of a structure at the same time, so it is ambiguous?